### ISSUE TITLE
`ClassDefBuilder.canAdd` is inconsistent with `ClassDefBuilder.add`

### ISSUE THREAD

**Issue by dscorbett** [fonttools/fonttools#2120]
If `ClassDefBuilder.canAdd` returns true for a certain glyph class, it is supposed to be safe to call `ClassDefBuilder.add` on that glyph class. However, `add` asserts that each glyph is not in `glyphs_` before adding it to `glyphs_`. Therefore, if a glyph class contains the same glyph twice, `canAdd` may return true but `add` will throw an error.

I noticed this because of the optimization in #2101.

**Comment by anthrotype** [comment]
Fixed by #2155

**Comment by dscorbett** [comment]
@anthrotype, I don’t think this has been fixed. #2155 improved the error message but did not address this bug.

**Comment by anthrotype** [comment]
oh ok. Then I didn't understand what the error is about :)
I'll try again when I get the chance
